cypress/integration/dashboard/dashboard_filtering_spec.js

```diff
@@ -0,0 +1,315 @@
+    })
+
+    it('tsvb top n is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbTopNValue"]', 2, /^0$/)
+    })
+
+    it('saved search is filtered', () => {
+      cy.checkElementDoesNotExist('[data-test-subj="docTableExpandToggleColumn"]')
+    })
+
+    // TODO: Uncomment once https://github.com/elastic/kibana/issues/22561 is fixed
+    // it('timeline is filtered', async () => {
+    //   await dashboardExpect.timelineLegendCount(0)
+    // })
+
+    it('vega is filtered', () => {
+      cy.get('.vgaVis__view text').each(($el, index, $list) => {
+        cy.get($el).should('not.contain', /^5,000$/)
+      })
+    })
+  })
+
+  describe('using a pinned filter that excludes all data', () => {
+    before(() => {
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-unpinned "]').click()
+      cy.get('[data-test-subj="pinFilter"]').click()
+    })
+
+    after(() => {
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-pinned "]').click()
+      cy.get('[data-test-subj="pinFilter"]').click()
+    })
+
+    it('filters on pie charts', () => {
+      // Check that none of the pie charts are occupied with data (show "No results found")
+      cy.checkElementDoesNotExist('svg > g > g.arcs > path.slice')
+    })
+
+    it('area, bar and heatmap charts filtered', () => {
+      // Check that none of the charts are filled with data
+      cy.checkElementDoesNotExist('svg > g > g.series')
+    })
+
+    it('data tables are filtered', () => {
+      // Check that none of the data tables are filled with data
+      cy.checkElementDoesNotExist('[data-test-subj="paginated-table-body"] [data-cell-content]')
+    })
+
+    it('goal and guages are filtered', () => {
+      // Goal label should be 0, gauge label should be 0%
+      cy.checkElementContainsValue('svg > g > g > text.chart-label', 2, /^0%?$/)
+    })
+
+    it('metric value shows no data', () => {
+      // The metrics should show '-'
+      cy.checkElementContainsValue('.mtrVis__value', 1, /^ - $/)
+    })
+
+    it('tag cloud values are filtered', () => {
+      cy.checkElementComponentDoesNotExist('[data-test-subj="tagCloudVisualization"]', 'svg > g > text')
+    })
+
+    it('tsvb metric is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbMetricValue"]', 1, /^0 custom template$/)
+    })
+
+    it('tsvb top n is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbTopNValue"]', 2, /^0$/)
+    })
+
+    it('saved search is filtered', () => {
+      cy.get('[data-test-subj="docTableExpandToggleColumn"]').should('not.exist')
+    })
+
+    // TODO: Uncomment once https://github.com/elastic/kibana/issues/22561 is fixed
+    // it('timeline is filtered', async () => {
+    //   await dashboardExpect.timelineLegendCount(0)
+    // })
```

Comment:
is it a flaky test? could we fix, or remove?

Reply:
Yes, it was a test that always failed due to the timeline visualization not updating as expected. I decided to remove all skipped tests associated with the timeline visualization.